core/src/SYCL/Kokkos_SYCL_ParallelReduce_MDRange.hpp

```diff
@@ -78,7 +78,7 @@ class Kokkos::Impl::ParallelReduce<CombinedFunctorReducerType,
         m_result_ptr_device_accessible(
             MemorySpaceAccess<Kokkos::Experimental::SYCLDeviceUSMSpace,
                               typename View::memory_space>::accessible),
-        m_shared_memory_lock(
+        m_host_scratch_lock(
```

Comment:
I don't get the new name.  Please explain.

Reply:
It guards access to scratch memory allocated in host-pinned/USM host space.

Reply:
Try `scratch_buffers_lock`

Reply:
b914bcf9e